tests/test_toa_reader.py

```diff
@@ -234,6 +235,51 @@ def test_toa_merge_different_ephem():
         nt = toa.merge_TOAs(toas)
 
 
+def test_toa_merge_different_columns():
+    filenames = [
+        datadir / "NGC6440E.tim",
+        datadir / "testtimes.tim",
+    ]
+    model = get_model(datadir / "NGC6440E.par")
+    # add a pulse_number column.  then the merge should fail
+    toas = [toa.get_TOAs(ff, model=model) for ff in filenames]
+    toas[0].compute_pulse_numbers(model)
+    with pytest.raises(TypeError):
```

Comment:
Can you check that the exception message contains the name of the offending column? Pytest provides some tools for asserting about exception messages.